### DISCUSSION TITLE
Apple Silicon toolchains 8.4.0+2021r2-patch3

### DISCUSSION THREAD

**Discussion by Jason2866** [espressif/arduino-esp32#6591] [Show and tell]
With the info provided here https://github.com/espressif/crosstool-NG/issues/22
i managed to backport the needed changes to compile `8.4.0+2021r2-patch3` on a Mac M1.
The toolchains are [here](https://github.com/Jason2866/crosstool-NG/releases) The source code is here:
https://github.com/Jason2866/crosstool-NG
https://github.com/Jason2866/espressif_gcc

**The performance gain is about 20-25%**

@me-no-dev @antmak Is it possible to upload to Platformio registry?
I already talked to @ivankravets about this.

**Comment by me-no-dev** [comment]
@Jason2866 I would suggest that you provide the changes to the crosstool-NG repo. It does not make sense to upload toolchains from outside the main repo releases. I will give them a shot though. Do you notice any difference?

**Comment by ivankravets** [comment]
@Jason2866, thanks again for your contributions! It would be great if you provide PR to the crosstool-NG repo.

I agree with @me-no-dev, we should avoid duplicated unofficial packages especially when they use custom versions, such as 8.5.0 that has not been released yet. We have just removed all duplicated packages. Please use [official packages in the PlatformIO Registry](https://registry.platformio.org/search?q=owner%3Aespressif) provided by Espressif Systems.

**Comment by Jason2866** [comment]
@me-no-dev  It is faster when compiling. The resulted generated code is the same. I will provide a PR against the espressif crosstool-NG and the gcc repo.

**Comment by Jason2866** [comment]
Opened the two needed PRs https://github.com/espressif/gcc/pull/1 https://github.com/espressif/crosstool-NG/pull/24
@me-no-dev
